tests/merkle_consistency/test_consistency.py

```diff
@@ -0,0 +1,93 @@
+"""
+run with
+$ pytest -v -s test_consistency
+"""
+
+from tqdm import tqdm
+import pytest
+from consistency import (
+    log2_ceiling,
+    closest_pow_of_2,
+    merkle_tree_hash,
+    path,
+    root_from_path,
+    cons_proof_sub,
+    root_0_from_const_proof,
+    root_1_from_const_proof,
+)
+
+
+@pytest.fixture
+def init_environment():
+    """
+    This runs before every test
+    """
+    global data
+    global start
+    global step
+    size = 33
+    start = 1
+    step = 1
+    data = []
+    for i in range(size):
+        data.append(int(i).to_bytes(32, "big"))
+
+
+def test_log2_ceiling(init_environment):
+
+    assert log2_ceiling(0)["result"] == 0
+    assert log2_ceiling(1)["result"] == 1
+    assert log2_ceiling(2)["result"] == 2
+    assert log2_ceiling(3)["result"] == 2
+    assert log2_ceiling(4)["result"] == 3
+    assert log2_ceiling(5)["result"] == 3
+    assert log2_ceiling(6)["result"] == 3
+    assert log2_ceiling(7)["result"] == 3
+    assert log2_ceiling(8)["result"] == 4
+
+
+def test_closest_pow_of_2(init_environment):
+
+    assert closest_pow_of_2(0)["result"] == 0
+    assert closest_pow_of_2(1)["result"] == 0
+    assert closest_pow_of_2(2)["result"] == 1
+    assert closest_pow_of_2(3)["result"] == 2
+    assert closest_pow_of_2(4)["result"] == 2
+    assert closest_pow_of_2(5)["result"] == 4
+    assert closest_pow_of_2(6)["result"] == 4
+    assert closest_pow_of_2(7)["result"] == 4
+    assert closest_pow_of_2(8)["result"] == 4
+    assert closest_pow_of_2(9)["result"] == 8
+    assert closest_pow_of_2(10)["result"] == 8
+
+
+def test_merkle_tree_hash(init_environment):
```

Comment:
can we get a property-based testing library and check compatibility with the mmr implementation we'll use on the prover? btw why did we even implement `path` in solidity?

Reply:
I will to look into hypothesis for property-based testing. But I am confused about mmrs, I thought they would be replaced by consistency proofs

Reply:
path is just for completeness

Reply:
consistency proofs are generated for MMRs, I don't understand how you mean they will be *replaced*

Reply:
the property based testing would be of the form

```
forall xs ([bytes]): python_library_mtr(xs) == solidity_mtr(xs)
```

To ensure correctness and compatibility between the two implementations.